src/main/kotlin/com/dansplugins/factionsystem/map/dynmap/DynmapService.kt

```diff
@@ -0,0 +1,298 @@
+        realm.groupBy { it.worldId }.forEach { (worldId, worldClaims) ->
+            taskScheduler.scheduleTask(faction.id, { updateWorldRealm(faction, worldId, worldClaims, realmMarkerSet) })
+        }
+    }
+
+    /**
+     * Updates the realm for the specified faction in the given world.
+     *
+     * @param faction The faction whose realm needs to be updated.
+     * @param worldId The ID of the world.
+     * @param worldClaims The list of claims in the world.
+     * @param realmMarkerSet The marker set for realms.
+     */
+    private fun updateWorldRealm(faction: MfFaction, worldId: UUID, worldClaims: List<MfClaimedChunk>, realmMarkerSet: MarkerSet) {
+        val world = plugin.server.getWorld(worldId)
+        if (world != null) {
+            taskScheduler.scheduleTask(faction.id, { createRealmMarkers(faction, world, worldClaims, realmMarkerSet) })
+        }
+    }
+
+    /**
+     * Creates realm markers for the specified faction in the given world.
+     *
+     * @param faction The faction whose realm markers need to be created.
+     * @param world The world.
+     * @param worldClaims The list of claims in the world.
+     * @param realmMarkerSet The marker set for realms.
+     */
+    private fun createRealmMarkers(faction: MfFaction, world: World, worldClaims: List<MfClaimedChunk>, realmMarkerSet: MarkerSet) {
+        val paths = claimPathBuilder.getPaths(worldClaims)
+        if (plugin.config.getBoolean("dynmap.debug")) {
+            plugin.logger.info("[Dynmap Service] Generated ${paths.size} paths for realm in world ${world.name}")
+        }
+        paths.forEachIndexed { index, path ->
+            val corners = getCorners(path)
+            taskScheduler.scheduleTask(faction.id, { createRealmAreaMarker(faction, world, corners, realmMarkerSet, index) })
+        }
+    }
+
+    /**
+     * Creates a realm area marker for the specified faction in the given world.
+     *
+     * @param faction The faction whose realm area marker needs to be created.
+     * @param world The world.
+     * @param corners The list of corner points.
+     * @param realmMarkerSet The marker set for realms.
+     * @param index The index of the realm area marker.
+     */
+    private fun createRealmAreaMarker(faction: MfFaction, world: World, corners: List<Point>, realmMarkerSet: MarkerSet, index: Int) {
+        val areaMarker = realmMarkerSet.createAreaMarker(
+            "realm_${faction.id}_${world.name}_$index",
+            faction.name,
+            false,
+            world.name,
+            corners.map { (x, _) -> x * 16.0 }.toDoubleArray(),
+            corners.map { (_, z) -> z * 16.0 }.toDoubleArray(),
+            false
+        )
+        areaMarker.label = faction.name
+        val color = Integer.decode(faction.flags[plugin.flags.color])
+        areaMarker.setFillStyle(0.0, color)
+        areaMarker.setLineStyle(4, 1.0, color)
+        areaMarker.description = FactionInfoBuilder(plugin).build(faction)
```

Comment:
The class instance of FactionInfoBuilder should be used here instead of instantiating an new instance in this method. The rest of the file should be checked for any more unnecessary instantiations

Reply:
Addressed